src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -559,9 +560,10 @@ def prepare_latents(
             else:
                 init_latents = self.vae.encode(image).latent_dist.sample(generator)
 
-            self.vae.to(dtype)
-            init_latents = init_latents.to(dtype)
+            if self.vae.config.force_upcast:
+                self.vae.to(dtype)
```

Comment:
If `force_upcast` is true, then shouldn't we use `upcast_vae()`?

Reply:
We have to possibly move it back here since for the decoder not all layers are upcasted (so we should move it back to fp16 here)